reposerver/repository/repository.go

```diff
@@ -1426,7 +1426,10 @@ func populateKustomizeAppDetails(res *apiclient.RepoAppDetailsResponse, q *apicl
 		kustomizeBinary = q.KustomizeOptions.BinaryPath
 	}
 	k := kustomize.NewKustomizeApp(appPath, q.Repo.GetGitCreds(), q.Repo.Repo, kustomizeBinary)
-	_, images, err := k.Build(q.Source.Kustomize, q.KustomizeOptions)
```

Comment:
I think kustomize should receive the same env variables as all other tools. Please use `newEnv` instead of creating env with one variable.

Reply:
Thank you for your review.

Yeah. I know I should use the `newEnv` defined [here](https://github.com/argoproj/argo-cd/blob/a17c77e238f44b442020e3779b8284879f5fb7f3/reposerver/repository/repository.go#L835). However it needs to pass the `apiclient.ManifestRequest` as the param, and there is no such variable in this [context](https://github.com/argoproj/argo-cd/blob/dfd6ec8e08a83f34259a9620a17761991e0fe711/reposerver/repository/repository.go#L1422-L1438). I don't want to change to much code at the first contribution. I thought maybe adding the `ARGOCD_APP_NAME` is also acceptable. 
Anyway, I can investigate more if you'd prefer to use the `newEnv` here also.

Reply:
It looks like I can construct the `ManifestRequest` with most of the necessary properties by myself without the `Namespace`. I don't know where to find the destination application namespace.
```
	mq := apiclient.ManifestRequest{
		AppName: q.AppName,
		Namespace: "",
		Repo: q.Repo,
		ApplicationSource: q.Source,
 	}
```

Reply:
This approach looks good. I would omit namespace for now. We can add the namespace to API in a separate PR